src/plugins/flow.js

```diff
@@ -1162,6 +1162,7 @@ export default function (instance) {
         let state = this.state.clone();
         try {
           let returnType = this.flowParseTypeAnnotation();
+          if (this.canInsertSemicolon()) this.unexpected();
```

Comment:
Can we make this maybe a nice error message here? Flow says "Illegal newline before arrow", which is already better than "unexpected token".